test/machines/awsclient.go

```diff
@@ -27,35 +27,60 @@ func (client *awsClientWrapper) GetRunningInstances(machine *clusterv1alpha1.Mac
 }
 
 func (client *awsClientWrapper) GetPublicDNSName(machine *clusterv1alpha1.Machine) (string, error) {
-	runningInstances, err := machineutils.GetRunningInstances(machine, client.client)
+	instance, err := machineutils.GetInstance(machine, client.client)
 	if err != nil {
 		return "", err
 	}
 
-	if len(runningInstances) == 0 {
-		return "", fmt.Errorf("no running machine instance found")
-	}
-
-	if *runningInstances[0].PublicDnsName == "" {
+	if *instance.PublicDnsName == "" {
 		return "", fmt.Errorf("machine instance public DNS name not set")
 	}
 
-	return *runningInstances[0].PublicDnsName, nil
+	return *instance.PublicDnsName, nil
 }
 
 func (client *awsClientWrapper) GetPrivateIP(machine *clusterv1alpha1.Machine) (string, error) {
-	runningInstances, err := machineutils.GetRunningInstances(machine, client.client)
+	instance, err := machineutils.GetInstance(machine, client.client)
```

Comment:
The same here. It's assumed the instance is running. So if nothing else, let's `s/GetPrivateIP/GetRunningInstancePrivateIP`.

Reply:
But this will break the `CloudProviderClient` interface.
Same as above, the `GetInstance` will be renamed to `GetRunningInstance`.